### PR TITLE
change update duties to handle all validators exited check

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Other

**What does this PR do? Why is it needed?**

Running validator status is a big strain on the system and it's only really needed at when validators are added/removed and on startup. Adding this check on the update duties which also contains the statuses anyways would be a better approach to handling this. This issue is more straining with many validators being pending status, especially for large operators.

### CODE REVIEW

File: validator/client/validator.go [hunk 1]
```diff
@@ -603,6 +604,16 @@ func (v *validator) UpdateDuties(ctx context.Context, slot primitives.Slot) erro
 		return err
 	}
 
+	allExitedCounter := 0
+	for i := range resp.CurrentEpochDuties {
+		if resp.CurrentEpochDuties[i].Status == ethpb.ValidatorStatus_EXITED {
+			allExitedCounter++
+		}
+	}
+	if allExitedCounter != 0 && allExitedCounter == len(resp.CurrentEpochDuties) {
```
Comment (terencechain) [inline validator/client/validator.go:613]: why do we need `allExitedCounter != 0`? 

`allExitedCounter == len(resp.CurrentEpochDuties)` seems sufficient

File: validator/client/runner.go [hunk 2]
```diff
@@ -110,7 +102,11 @@ func run(ctx context.Context, v iface.Validator) {
 			// Keep trying to update assignments if they are nil or if we are past an
 			// epoch transition in the beacon node's state.
 			if err := v.UpdateDuties(ctx, slot); err != nil {
-				handleAssignmentError(err, slot)
+				if errors.Is(err, ErrValidatorsAllExited) {
```
Comment (terencechain) [inline validator/client/runner.go:105]: Why not have this under `handleAssignmentError`? It already handles pending validators anyway. It seems cleaner to push this responsibility to `handleAssignmentError` under a case state ment for pending, exit, and everything else

File: validator/client/validator.go (hunk 1 above)
Comment (james-prysm) [reply validator/client/validator.go:613]: if they are both equal to 0 it'd pass, I guess it's just not displaying that log if there's no keys

File: validator/client/runner.go (hunk 2 above)
Comment (james-prysm) [reply validator/client/runner.go:105]: I was thinking that initially I'm ok either way i suppose, thought handle assignment error dealt with some other things but i'll move it.

**Pull request merged**